CONTRIBUTING.md

```diff
@@ -0,0 +1,241 @@
+about how and when to split changes.
+
+#### Dependency Management
+
+When submitting a PR that has dependencies in other repositories these
+dependencies should be stated in the PR or the commits using the `Depends-On:`
+tag.
+
+There are 4 ways to state a dependency with another projects PR:
+- `Depends-On: lib-common=88`
+- `Depends-On: lib-common#88`
+- `Depends-On: openstack-k8s-operators/lib-common#88`
+- `Depends-On: https://github.com/openstack-k8s-operators/lib-common/88`
+
+Multiple `Depends-On:` tags are supported.
+
+For the time being these tags are only useful when using the
+`hack/checkout_pr.sh` or `hack/showdeps.py` scripts.
+
+A good example of using these tags is the `extraVol` series of PRs. There are
+PRs in 4 projects: lib-common, cinder-operator, glance-operator, and the
+openstack-operator.
+
+The operators all require the lib-common PR, but then there are 2 circular
+requirements.  One is between the cinder-operator and the openstack-operator,
+and the other is between the glance-operator and the openstack-operator.
+
+These are the PRs for reference:
+
+- https://github.com/openstack-k8s-operators/lib-common/pull/88
+- https://github.com/openstack-k8s-operators/cinder-operator/pull/65
+- https://github.com/openstack-k8s-operators/glance-operator/pull/75
+- https://github.com/openstack-k8s-operators/openstack-operator/pull/38
+
+#### Structural split of changes
+
+The general rule for how to split code changes into commits is that we should
+aim to have a single "logical change" per commit.
+
+As the [OpenStack Git Commit Good
+Practice](https://wiki.openstack.org/wiki/GitCommitMessages) explains, there
+are multiple reasons why this rule is important:
+
+- The smaller the amount of code being changed in a commit, the quicker &
+  easier it is to review & identify potential flaws in that specific code
+  change.
+- If a change is found to be flawed later, it may be necessary to revert the
+  broken commit. This is much easier to do if there are not other unrelated
+  code changes entangled with the original commit.
+- When troubleshooting problems using Git's bisect capability, small well
+  defined changes will aid in isolating exactly where the code problem was
+  introduced.
+- When browsing history using Git annotate/blame, small well defined changes
+  also aid in isolating exactly where & why a piece of code came from.
+
+In the [OpenStack Git Commit Good Practice
+page](https://wiki.openstack.org/wiki/GitCommitMessages) we can also find two
+great sections explaining [things to avoid when creating
+commits](https://wiki.openstack.org/wiki/GitCommitMessages#Things_to_avoid_when_creating_commits)
+and [examples of bad
+practice](https://wiki.openstack.org/wiki/GitCommitMessages#Examples_of_bad_practice)
+that contributors to this repository should take into consideration.
+
+#### PR Images
+
+Once a PR merges it will trigger an image rebuild and publish, so how can we
```

Comment:
Good question. I don't know but I'm commenting here to bring attention to other reviewers who might know. @dprince do you know?